hw/drivers/flash/spiflash/src/spiflash.c

```diff
@@ -1130,12 +1131,13 @@ spiflash_execute_erase(struct spiflash_dev *dev, const uint8_t *buf,
     /* Now we know that device is not ready */
     dev->ready = false;
 
-    start_time = os_cputime_ticks_to_usecs(os_cputime_get32());
+    start_time = os_cputime_get32();
     /* Wait typical erase time before starting polling for ready */
     spiflash_delay_us(delay_spec->typical);
 
     /* Poll status ready for remaining time */
-    wait_time_us = start_time + delay_spec->maximum - os_cputime_get32();
+    wait_time_us = delay_spec->maximum -
```

Comment:
Is there any chance that the time has advanced such that this calculation "goes negative" and thus ends up with a super high wait time?

Changes look good otherwise.

Reply:
Your are right, it could happen when delay times were very short, specially when maximum time was shorter then typical time.

Reply:
I think I might be a bit confused with the force push but will make comments on what I think the changes were. For some reason I do not like the typecast of the unit32_t timeout to an int32_t to check if it is negative. I would have made sure that wait_time_us did not go negative. If anything, I would check the input timeout_us to see if it is too large but keep it uint32_t (not a check for negative).

Reply:
I'm sorry about force push, but change is simple enough.